File: 3090. Maximum Length Substring With Two Occurrences.java
class Solution {
    public int maximumLengthSubstring(String s) {
        int max = 0;
        for (int i = 0; i < s.length(); i++) {
            max = Math.max(find(s, i), max);
        }
        return max;
    }
    private int find(String s, int start) {
        int[] freq = new int[26];
        int length = 0;
        for (int i = start; i < s.length(); i++) {
            if (freq[s.charAt(i)-'a'] == 2) return length;
            freq[s.charAt(i)-'a']++;
            length++;
        }
        return length;
    }
}

File: 3090. Maximum Length Substring With Two Occurrences.go
func maximumLengthSubstring(s string) int {
    answer := 0
    for i, _ := range(s) {
        answer = max(find(s, i), answer)
    }
    return answer
}
func find(s string, start int) int {
    freq := [26]int{}
    length := 0
    for i := start; i < len(s); i++ {
        if freq[s[i]-'a'] == 2 {
            return length
        }
        freq[s[i]-'a'] += 1
        length += 1
    }
    return length
}

